### PR TITLE
8293781: RISC-V: Clarify types of calls

### PR DESCRIPTION
**Pull request by yhzhu20**
RISCV64 OpenJDK uses different types of calls in generated calls. It should be clarified what they are. And the PR also cleans up some out-of-date comments.

HotSpot tier1 tests passed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293781](https://bugs.openjdk.org/browse/JDK-8293781): RISC-V: clarify types of calls

### Reviewers
 * [Feilong Jiang](https://openjdk.org/census#fjiang) (@feilongjiang - Author)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Yadong Wang](https://openjdk.org/census#yadongwang) (@yadongw - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10311/head:pull/10311` \
`$ git checkout pull/10311`

Update a local copy of the PR: \
`$ git checkout pull/10311` \
`$ git pull https://git.openjdk.org/jdk pull/10311/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10311`

View PR using the GUI difftool: \
`$ git pr show -t 10311`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10311.diff">https://git.openjdk.org/jdk/pull/10311.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/riscv/macroAssembler_riscv.hpp [hunk 1]
```diff
@@ -635,6 +637,69 @@ class MacroAssembler: public Assembler {
...
+  //     the range [-(2G + 2K), 2G - 2K). Addresses out of the range in the code cache
+  //     requires indirect call.
+  //     If a jump is needed rather than a call, a far jump 'j reg' can be used instead.
+  //     All instructions are embedded at a call site.
+  //
+  //   - trampoline call:
+  //     This is only available in C1/C2-generated code (nmethod). It is a combination
+  //     of a direct call, which is used if the destination of a call is in range,
+  //     and a register-indirect call. It has the advantages of reaching anywhere in
+  //     the AArch64 address space and being patchable at runtime when the generated
```
Comment (feilongjiang) [inline src/hotspot/cpu/riscv/macroAssembler_riscv.hpp:655]: s/AArch64/RISC-V

**Review by feilongjiang — CHANGES_REQUESTED** [review]
Looks good, with one comment:

File: src/hotspot/cpu/riscv/macroAssembler_riscv.hpp (hunk 1 above)
Comment (yhzhu20) [reply src/hotspot/cpu/riscv/macroAssembler_riscv.hpp]: Done.

**Review by RealFYang — APPROVED** [review]
Looks good. Thanks.

**Review by feilongjiang — APPROVED** [review]
Change looks good.

**Review by yadongw — APPROVED** [review]
lgtm(not a reviewer)

**Pull request closed without merging**